pkg/builder/project.go

```diff
@@ -91,6 +94,16 @@ func generateJavaKeystore(ctx *builderContext) error {
 	return jvm.GenerateKeystore(ctx.C, ctx.Path, ctx.Maven.TrustStoreName, ctx.Maven.TrustStorePass, certsData)
 }
 
+func mergeSecrets(secrets []corev1.SecretKeySelector, secret *corev1.SecretKeySelector) []corev1.SecretKeySelector {
+	if secrets == nil && secret == nil {
```

Comment:
Is this check necessary?  `secrets == nil`
I think it's always safe to assume a slice is not nil in Go.

Reply:
Hi @tadayosi ! Thanks for reviewing! 

I didn't know a slice cannot be nil in Go ? 

https://go.dev/tour/moretypes/12#:~:text=The%20zero%20value%20of%20a,and%20has%20no%20underlying%20array.

I just wrote it to be bullet proof but I am happy to remove it if you think it's unnecessary ?

Thanks !

Reply:
After running it debug mode it seems that `ctx.Build.Maven.CASecrets` does get set to nil so the check is necessary ? Thanks!

Reply:
Let me merge this for now and we can reiterate later